packages/squiggle-lang/src/js/index.ts

```diff
@@ -99,25 +104,74 @@ export type squiggleExpression =
 export function run(
   squiggleString: string,
   bindings?: externalBindings,
-  samplingInputs?: samplingParams
+  samplingInputs?: samplingParams,
+  parameters?: parameters
 ): result<squiggleExpression, errorValue> {
-  let b = bindings ? bindings : {};
+  let b = bindings ? bindings : defaultBindings;
+  let p = parameters ? parameters : defaultParameters;
   let si: samplingParams = samplingInputs
     ? samplingInputs
     : defaultSamplingInputs;
 
   let result: result<expressionValue, errorValue> =
-    evaluateUsingExternalBindings(squiggleString, b);
+    evaluateUsingExternalBindings(squiggleString, mergeParameters(b, p));
   return resultMap(result, (x) => createTsExport(x, si));
 }
 
 // Run Partial. A partial is a block of code that doesn't return a value
 export function runPartial(
   squiggleString: string,
-  bindings: externalBindings,
-  _samplingInputs?: samplingParams
+  bindings?: externalBindings,
+  _samplingInputs?: samplingParams,
+  parameters?: parameters
 ): result<externalBindings, errorValue> {
-  return evaluatePartialUsingExternalBindings(squiggleString, bindings);
+  let b = bindings ? bindings : defaultBindings;
+  let p = parameters ? parameters : defaultParameters;
+
+  return evaluatePartialUsingExternalBindings(
+    squiggleString,
+    mergeParameters(b, p)
+  );
+}
+
+function mergeParameters(
+  bindings: externalBindings,
+  parameters: parameters
+): externalBindings {
+  let transformedParemeters = Object.fromEntries(
+    Object.entries(parameters).map(([key, value]) => [
+      "$" + key,
```

Comment:
We should probably soon have some validation on these, though that could be a separate story. Like, we enforce that people can't already have "$" in the name. (And can't do this within Squiggle proper).